4_naming_conventions/42_file_names.md

```diff
@@ -58,3 +58,127 @@ regular expression:
 
 That is, a letter followed by zero, or more, letters, underscores, dashes, or
 digits followed by a period followed by one or more letters or digits.
+
+### 4.2.5 File naming guidelines for modules
+
+Below sections are the file naming guidelines for EDK II meta files and source files.
+
+#### 4.2.5.1 EDK II meta files within a package
+
+```
+<PackageName>Pkg.dec
+<PackageName>Pkg.dsc
+
+   <PackageName> REQUIRED  *
+```
+
+#### 4.2.5.2 EDK II INF file within a Module instance
+* If the implementation supports >=2 or all CPU archs:
+```
+<ModuleName><Phase>.inf
+
+   <ModuleName>   REQUIRED    *
+   <Phase>        REQUIRED    Base, Sec, Pei, Dxe, DxeRuntime, Mm, StandaloneMm,
+                              Smm, Uefi.
+
+Example:
+   CpuIo2Dxe.inf
+```
+

```

Comment:
We have to be careful here.  Just because the very first implementation of a component may be for a single CPU or CPU+Vendor, that does not mean that additional CPU/Vendor support may not be added in the future.  We do not want to change the directory of INF name later because that breaks backwards compatibility.  The rule below must only be used if the developer can not imagine any other CPU/Vendor combinations every being added later

Reply:
We probably can mention "If the implementation supports >=2 or all CPU archs" case on the wiki page. That is ok we don't assign <arch> or <vendor> when the first time to implement the module. We won't rename the directory or INF. We will create a new INF file for arch or vendor instead.
Yeah, seems to me <ModuleName><Phase><CpuArch><Vendor>.inf covers <ModuleName><Phase>.inf case.